File: Go/gazasort.go
package main

import (
    "math/rand"
    "time"
)

func gazaSort(arr []int) []int {
    for !isSorted(arr) {
        rand.Seed(time.Now().UnixNano())
        arr = append(arr[:rand.Intn(len(arr))], arr[rand.Intn(len(arr))+1:]...)
    }
    return arr
}

func isSorted(arr []int) bool {
    for i := 1; i < len(arr); i++ {
        if arr[i-1] > arr[i] {
            return false
        }
    }
    return true
}

File: Rust/gazasort.rs
use rand::Rng;

fn gaza_sort(mut arr: Vec<i32>) -> Vec<i32> {
    while !is_sorted(&arr) {
        let mut rng = rand::thread_rng();
        let idx = rng.gen_range(0..arr.len());
        arr.remove(idx);
    }
    arr
}

fn is_sorted(arr: &[i32]) -> bool {
    arr.windows(2).all(|w| w[0] <= w[1])
}
